src/Library/Plugin/Notice.php

```diff
@@ -0,0 +1,295 @@
+            $this->id        = $notice['id'];
+            $this->pageSlug  = $notice['page']; 
+            $this->version   = $notice['version'];
+            $this->isUnread  = true;
+            $this->plugin    = $plugin;
+        }
+    }
+
+	/**
+	 * Get notice ID
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getId() {
+        return $this->id;
+    }
+
+	/**
+	 * Set notice ID
+	 *
+	 * @param string $id Notice ID.
+     * @access private
+	 * @since SINCEVERSION
+	 *
+	 */
+	private function setId( $id ) {
+		$this->id = $id;
+	}
+
+	/**
+	 * Get slug of the page this feature is on.
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getPageSlug() {
+        return $this->pageSlug;
+    }
+
+	/**
+	 * Set pageSlug
+	 *
+	 * @param string $pageSlug Notice Page.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setPageSlug( string $pageSlug ) {
+		$this->pageSlug = $pageSlug;
+	}
+
+	/**
+	 * Get version of plugin this notice was added on
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getVersion() {
+        return $this->version;
+    }
+
+	/**
+	 * Set version of plugin this notice was added on
+	 *
+	 * @param string $version Version of plugin this notice was added on.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setVersion( string $version ) {
+		$this->version = $version;
+	}
+
+    /**
+     * Maybe Show
+     * 
+     * Returns true if this plugin's version number is
+     * greater than the first installed version, and feature's version number is 
+     * greater than or equal to current version
+     * @return bool
+     * @since SINCEVERSION
+     */
+    
+    public function maybeShow() {
+        $pluginVersion = $this->plugin->getPluginData()['Version'];
+        $versionIsNotFirst = $this->plugin->firstVersionCompare( $pluginVersion, '<' );
+        $featureIsNewer = version_compare( $this->version, $pluginVersion , '>=' );
+        return ( $featureIsNewer && $versionIsNotFirst);
+    }
+
+	/**
+	 * Get isUnread Value.
+	 *
+	 * @return bool
+	 * @since SINCEVERSION
+	 */
+    public function getIsUnread() {
+        if ( $this->maybeShow() ) {
+            return $this->isUnread;
+        }
+        return false;
+    }
+
+	/**
+	 * Set the notice as read or unread
+	 *
+	 * @param bool $isUnread Specifies if the Notice is Unread or not.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setIsUnread( $isUnread ) {
+        $this->isUnread = $isUnread;
+        $this->updateNoticeOption();
+    }
+    
+    /**
+     * Determine if Notice Exists in Options Table
+     * 
+     * If the notice exists, then the return value is the index
+     * of that notice in the $option array. If the notice does not exist
+     * this returns false.
+     * 
+     * @return mixed 
+     * @since SINCEVERSION
+     */
+    public function alreadyExists( $noticeId ) {
+        $option = get_option( 'boldgrid_plugin_page_notices');
```

Comment:
If the option does not exist, it will return false and when you `foreach ( $option as $notice )` you will get a warning about using a `foreach` with a bool.

You can pass a default value so you always get an array back:
```
$option = get_option( 'boldgrid_plugin_page_notices', [] );
```

Doing that will eliminate the need for `if ( $option )`